conjure-core/src/main/java/com/palantir/conjure/parser/LogSafetyDefinition.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * (c) Copyright 2022 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.parser;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonCreator.Mode;
+import com.palantir.conjure.exceptions.ConjureIllegalArgumentException;
+import com.palantir.logsafe.Preconditions;
+
+public enum LogSafetyDefinition {
+    SAFE("safe"),
+    UNSAFE("unsafe"),
+    DO_NOT_LOG("do-not-log");
+
+    private final String name;
+
+    LogSafetyDefinition(String name) {
+        this.name = name;
+    }
+
+    @JsonCreator(mode = Mode.DELEGATING)
+    public static LogSafetyDefinition parse(String value) {
```

Comment:
Why this extra class instead of directly parsing to `LogSafety`? Is it to support the more human-friendly values (e.g. "do-not-log" instead of "DO_NOT_LOG")?

Reply:
That’s correct, we have two apis: the human interaction api (directly parses yaml) and the more structured conjure-defined intermediate representation. This class handles the yaml model where we want to provide consistency with other elements (lower kebab case)